src/Text/Heterocephalus.hs

```diff
@@ -486,6 +486,17 @@ docToExp set scope (DocCond conds final) = do
       let d' = derefToExp ((specialOrIdent, VarE 'or) : scope) d
       docs' <- docsToExp set scope docs
       return $ TupE [d', docs']
+docToExp set scope (DocCase deref cases) = do
+    let exp_ = derefToExp scope deref
+    matches <- mapM toMatch cases
+    return $ CaseE exp_ matches
+  where
+    toMatch :: (Binding, [Doc]) -> Q Match
+    toMatch (idents, inside) = do
+        (pat, extraScope) <- bindingPattern idents
+        let scope' = extraScope ++ scope
```

Comment:
I don't yet understand %100 how `Scope` works with heterocephalus/shakespeare.

Does this look alright, or does something need to be changed for heterocephalus?

I copied this branch of the `docToExp` verbatim from `Text.Hamlet`.